gov/politeia/proposals.go

```diff
@@ -214,29 +214,29 @@ func (db *ProposalDB) saveProposals(publicProposals pitypes.Proposals) (int, err
 		// that generates a RefID similar to one already in the db and appending
 		// integers to it till it becomes unique is the solution.
 		if err == storm.ErrAlreadyExists {
+			var data *pitypes.ProposalInfo
 			// Check if the proposal token already exists in the db.
-			data, newErr := db.proposal("TokenVal", val.TokenVal)
-			if newErr == nil && data != nil {
+			data, err = db.proposal("TokenVal", val.TokenVal)
+			if err == nil && data != nil {
 				// The proposal token already exists thus trigger an update with
 				// the latest details.
 				valCopy := *val
 				valCopy.ID = data.ID
-				prefixStr := ""
+				suffixStr := ""
 
 				for k := 1; k <= maxLoop; k++ {
-					valCopy.RefID += prefixStr
+					valCopy.RefID += suffixStr
 					// Attempt to update the old entry.
-					newErr = db.dbP.Update(&valCopy)
-					if newErr == nil || newErr != storm.ErrAlreadyExists {
-						err = nil
+					err = db.dbP.Update(&valCopy)
+					if err == nil || err != storm.ErrAlreadyExists {
```

Comment:
The issue I was pointing out still exists.  Consider `err != storm.ErrAlreadyExists`.  this is true if `err` is `nil` (assuming `storm.ErrAlreadyExists` is not equal to `nil`).  Thus, the `err == nil` part of the condition is not necessary.

Also, you still want to catch (and log or whatever) legit errors.

```go
if err == storm.ErrAlreadyExists {
	suffixStr = strconv.Itoa(k)
	continue
}
if err != nil {
	log.Error("storm DB Update failed: %v", err)
	// do something else?  return err?
}
break
```